### DISCUSSION TITLE
how do I disable multiprocessing?

### DISCUSSION THREAD

**Discussion by alita-moore** [albumentations-team/albumentations#2184] [Q&A] [answered]
as far as I can tell albumentations automatically performs multiprocessing, but I want to run one augmentation per cpu core (I want to manage the multiprocessing myself). Is it possible to disable multiprocessing for albumentations?

**Comment by ternaus** [comment, marked as answer]
Albumentations by itself does not perform multiprocessing, but libraries it depends on could, like opencv or numpy.

```python
import os
os.environ["OMP_NUM_THREADS"] = "1"
os.environ["OPENBLAS_NUM_THREADS"] = "1"
os.environ["MKL_NUM_THREADS"] = "1"
os.environ["VECLIB_MAXIMUM_THREADS"] = "1"
os.environ["NUMEXPR_NUM_THREADS"] = "1"
```
should do the job.

**Reply by alita-moore** [reply]
setting these variables globally when calling the script i.e. `OMP_NUM_THREADS=1 python example.py` worked
